libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs

```diff
@@ -251,10 +320,43 @@ private string EvalMultiLineText(string exp)
         {
             // remove ``` ```
             exp = exp.Substring(3, exp.Length - 6);
-            var reg = @"@\{[^{}]+\}";
-            var evalutor = new MatchEvaluator(m => EvalExpression(m.Value));
+            return EvalTextContainsExpression(exp);
+        }
+
+        private string EvalTextContainsExpression(string exp)
+        {
+            var evalutor = new MatchEvaluator(m => EvalExpression(m.Value).ToString());
+            return Regex.Replace(exp, ExpressionRecognizePattern, evalutor);
```

Comment:
>Regex.Rep [](start = 19, length = 9)

You should not call Regex.Replace or Regex.Match() directly.

it is 1000x more performant to create a regex object and cache it and reuse it, as it will turn expression into dynamically compiled code behind the scenes.

var regex = new Regex("....", RegexOptions.Compiled);

Reply:
Thanks for reminding, let me learn new knowledge.^_^